xbmc/LangInfo.cpp

```diff
@@ -275,6 +275,14 @@ void CLangInfo::CRegion::SetGlobalLocale()
 #endif
   }
 
+  try
+  {
+    g_langInfo.m_originalLocale = std::locale(strLocale.c_str());
+  }
+  catch (...) {
```

Comment:
I'm assuming there's an error if this happen.. You should catch the specific exception, and log it.